echo-rest/src/test/java/RestEventListenerTest.java

```diff
@@ -0,0 +1,291 @@
+        listener.getMapper().convertValue(event, new TypeReference<>() {});
+
+    listener.processEvent(event);
+
+    Mockito.verify(restService, Mockito.times(1)).recordEvent(expectedEvent);
+  }
+
+  @Test
+  void sendsEventsToMultipleHosts() {
+    // Given
+    RestService restService2 = Mockito.mock(RestService.class);
+
+    RestProperties.RestEndpointConfiguration config =
+        new RestProperties.RestEndpointConfiguration();
+    config.setWrap(false);
+
+    RestUrls.Service service1 =
+        RestUrls.Service.builder().client(restService).config(config).build();
+
+    RestUrls.Service service2 =
+        RestUrls.Service.builder().client(restService2).config(config).build();
+
+    listener.getRestUrls().setServices(List.of(service1, service2));
+
+    Map<String, Object> expectedEvent =
+        listener.getMapper().convertValue(event, new TypeReference<>() {});
+
+    listener.processEvent(event);
+
+    Mockito.verify(restService, Mockito.times(1)).recordEvent(expectedEvent);
+    Mockito.verify(restService2, Mockito.times(1)).recordEvent(expectedEvent);
+  }
+
+  @Test()
+  void exceptionInSendingEventToOneHostDoesNotAffectSecondHost() {
+    RestService restService2 = Mockito.mock(RestService.class);
+
+    RestProperties.RestEndpointConfiguration config =
+        new RestProperties.RestEndpointConfiguration();
+    config.setWrap(false);
+    config.setRetryCount(3);
+
+    RestUrls.Service service1 =
+        RestUrls.Service.builder().client(restService).config(config).build();
+
+    RestUrls.Service service2 =
+        RestUrls.Service.builder().client(restService2).config(config).build();
+
+    listener.getRestUrls().setServices(List.of(service1, service2));
+
+    Map<String, Object> expectedEvent =
+        listener.getMapper().convertValue(event, new TypeReference<>() {});
+
+    Mockito.when(restService.recordEvent(expectedEvent)).thenThrow(new RuntimeException());
+
+    listener.processEvent(event);
+
+    Mockito.verify(restService, Mockito.times(3)).recordEvent(expectedEvent);
+    Assertions.assertThrows(RuntimeException.class, () -> restService.recordEvent(expectedEvent));
+    Mockito.verify(restService2, Mockito.times(1)).recordEvent(expectedEvent);
+  }
+
+  @Test
+  void shouldSendEventWhenCircuitBreakerIsEnabled() {
+    RestProperties.RestEndpointConfiguration config =
+        new RestProperties.RestEndpointConfiguration();
+
+    RestUrls.Service service =
+        RestUrls.Service.builder().client(restService).config(config).build();
+
+    listener.setCircuitBreakerEnabled(true);
+    listener.getRestUrls().setServices(List.of(service));
+
+    Map<String, Object> expectedEvent =
+        listener.getMapper().convertValue(event, new TypeReference<>() {});
+
+    listener.processEvent(event);
+
+    Mockito.verify(restService, Mockito.times(1)).recordEvent(expectedEvent);
+  }
+
+  @Test
+  void shouldNotInvokeMultipleTimesAfterExceptionWhenCircuitBreakerEnabled() {
```

Comment:
Can you include a test where an exception is thrown in the `transformEventToMap`method?

Reply:
Added a couple of tests for this.